### PR TITLE
CI debug

### LINKED ISSUE #27541: CI: failing
https://travis-ci.org/pandas-dev/pandas/builds/562549143

```
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/_pytest/main.py", line 213, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/_pytest/main.py", line 257, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/hooks.py", line 289, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/manager.py", line 87, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/manager.py", line 81, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "<remote exec>", line 67, in pytest_runtestloop
INTERNALERROR>   File "<remote exec>", line 79, in run_one_test
INTERNALERROR> IndexError: list index out of range
[gw2] node down: Traceback (most recent call last):
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_base.py", line 1374, in _save
    dispatch = self._dispatch[tp]
KeyError: <enum 'ExitCode'>
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_base.py", line 1063, in executetask
    do_exec(co, loc)  # noqa
  File "<string>", line 1, in do_exec
  File "<remote exec>", line 249, in <module>
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/hooks.py", line 289, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/manager.py", line 87, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/manager.py", line 81, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/_pytest/main.py", line 250, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/_pytest/main.py", line 243, in wrap_session
    session=session, exitstatus=session.exitstatus
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/hooks.py", line 289, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/manager.py", line 87, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/manager.py", line 81, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "<remote exec>", line 46, in pytest_sessionfinish
  File "<remote exec>", line 31, in sendevent
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_base.py", line 708, in send
    self.gateway._send(Message.CHANNEL_DATA, self.id, dumps_internal(item))
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_base.py", line 1345, in dumps_internal
    return _Serializer().save(obj)
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_base.py", line 1363, in save
    self._save(obj)
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_base.py", line 1381, in _save
    dispatch(self, obj)
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_base.py", line 1466, in save_tuple
    self._save(item)
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_base.py", line 1381, in _save
    dispatch(self, obj)
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_base.py", line 1462, in save_dict
    self._write_setitem(key, value)
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_base.py", line 1456, in _write_setitem
    self._save(value)
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_base.py", line 1381, in _save
    dispatch(self, obj)
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_base.py", line 1462, in save_dict
    self._write_setitem(key, value)
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_base.py", line 1456, in _write_setitem
    self._save(value)
  File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_base.py", line 1379, in _save
    raise DumpError("can't serialize {}".format(tp))
execnet.gateway_base.DumpError: can't serialize <enum 'ExitCode'>
fReplacing crashed worker gw2
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/_pytest/main.py", line 213, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/_pytest/main.py", line 257, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/hooks.py", line 289, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/manager.py", line 87, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/manager.py", line 81, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/callers.py", line 203, in _multicall
INTERNALERROR>     gen.send(outcome)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/xdist/dsession.py", line 115, in pytest_runtestloop
INTERNALERROR>     self.loop_once()
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/xdist/dsession.py", line 138, in loop_once
INTERNALERROR>     call(**kwargs)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/xdist/dsession.py", line 207, in worker_errordown
INTERNALERROR>     self._clone_node(node)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/xdist/dsession.py", line 290, in _clone_node
INTERNALERROR>     node = self.nodemanager.setup_node(spec, self.queue.put)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/xdist/workermanage.py", line 68, in setup_node
INTERNALERROR>     gw = self.group.makegateway(spec)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/multi.py", line 127, in makegateway
INTERNALERROR>     io = gateway_io.create_io(spec, execmodel=self.execmodel)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_io.py", line 117, in create_io
INTERNALERROR>     return Popen2IOMaster(args, execmodel)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_io.py", line 20, in __init__
INTERNALERROR>     self.popen = p = execmodel.PopenPiped(args)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/site-packages/execnet/gateway_base.py", line 169, in PopenPiped
INTERNALERROR>     return self.subprocess.Popen(args, stdout=PIPE, stdin=PIPE)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/subprocess.py", line 729, in __init__
INTERNALERROR>     restore_signals, start_new_session)
INTERNALERROR>   File "/home/travis/miniconda3/envs/pandas-dev/lib/python3.6/subprocess.py", line 1295, in _execute_child
INTERNALERROR>     restore_signals, start_new_session, preexec_fn)
INTERNALERROR> OSError: [Errno 12] Cannot allocate memory
```

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/27541

### CODE REVIEW

**Comment by WillAyd** [comment]
Different error this time right?

**Comment by TomAugspurger** [comment]
We did have a timeout earlier, but I think it looked a bit different than the last one. I restarted the build.

File: ci/deps/travis-36-cov.yaml [hunk 1]
```diff
@@ -40,7 +40,7 @@ dependencies:
   - xlwt
   # universal
   - pytest>=4.0.2
```
Comment (jreback) [inline ci/deps/travis-36-cov.yaml]: you need pytest<5.0.0; I *thought* we had added this (its on all the other 3.6 builds)

**Comment by WillAyd** [comment]
I don't know if directly relevant here but there is a note for pytest >5.0.0 that if used with pytest-xdist that should at least be 1.29.0 or later in the changeling:

https://docs.pytest.org/en/latest/changelog.html#features

Might be related to some of the issues (this one seems suspiciously related to the one we saw last week on Azure)

**Comment by WillAyd** [comment]
Though it does look like the issue Jeff has in the OP of #27541 has pytest-xdist 1.28.0 and pytest 5.0.1 so issue might just boil down to that

**Comment by WillAyd** [comment]
Ignore comments above - I should have looked at diff. But yea this seems like the right change

**Comment by TomAugspurger** [comment]
Huh that sounds extremely relevant. In a meeting now, but will update everything to be pytest >5, pytest-xdist >=1.2.9 (unless someone beats me).

**Comment by WillAyd** [comment]
Current build here meets those specs but just failed again so I think it's still something else

**Comment by TomAugspurger** [comment]
Trying to pin that build to `pytest<5`. Won't have more time for this today unfortunately.

**Comment by TomAugspurger** [comment]
Well this one failed with out of memory errors when we tried to open a subprocess

```
=================================== FAILURES ===================================
_____________________________ test_initial_warning _____________________________
[gw0] linux -- Python 3.6.8 /home/travis/miniconda3/envs/pandas-dev/bin/python
    def test_initial_warning():
        code = (
            "import pandas as pd; import matplotlib.pyplot as plt; "
            "s = pd.Series(1, pd.date_range('2000', periods=12)); "
            "fig, ax = plt.subplots(); "
            "ax.plot(s.index, s.values)"
        )
        call = [sys.executable, "-c", code]
>       out = subprocess.check_output(call, stderr=subprocess.STDOUT).decode()

```

**Comment by WillAyd** [comment]
Yea was about to comment. I have the feeling this might be the root issue and the original issues we saw were problems with pytest / pytest-xdist compat and actually reporting out the error.

**Comment by jbrockmendel** [comment]
Is it this same test failing repeatedly?  We could wrap the subprocess call in something like:

```
try:
     subprocess...
except OSError as err:
     if err.errno == errno.ENOMEM:
          pytest.xfail(...)
     raise
```

**Comment by TomAugspurger** [comment]
We do seem to have less *disk space*, which indicates we may be running on a smaller VM. From https://travis-ci.org/pandas-dev/pandas/jobs/561486224 (3 days ago)

```
[0K$ df -h
Filesystem      Size  Used Avail Use% Mounted on
udev            3.7G  4.0K  3.7G   1% /dev
tmpfs           748M  292K  748M   1% /run
/dev/sda1        69G   11G   55G  17% /
none            4.0K     0  4.0K   0% /sys/fs/cgroup
none            5.0M     0  5.0M   0% /run/lock
none            3.7G     0  3.7G   0% /run/shm
none            100M     0  100M   0% /run/user
none            768M  179M  590M  24% /var/ramfs
```

And from today

```
[0K$ df -h
Filesystem      Size  Used Avail Use% Mounted on
udev            1.8G  4.0K  1.8G   1% /dev
tmpfs           370M  292K  370M   1% /run
/dev/sda1        69G   12G   55G  17% /
none            4.0K     0  4.0K   0% /sys/fs/cgroup
none            5.0M     0  5.0M   0% /run/lock
none            1.9G     0  1.9G   0% /run/shm
none            100M     0  100M   0% /run/user
none            768M  179M  590M  24% /var/ramfs
```

File: .travis.yml [hunk 2]
```diff
@@ -1,4 +1,3 @@
-sudo: false
```
Comment (WillAyd) [inline .travis.yml]: Shot in the dark but looks like this was deprecated

https://docs.travis-ci.com/user/reference/trusty/#container-based-infrastructure

**Comment by TomAugspurger** [comment]
FYI, we may want to keep an eye on https://travis-ci.community/t/builds-often-fail-because-of-lack-of-resources-memory-cpu/4331 and https://github.com/travis-ci/worker/issues/604. Seems like the worker images are being given less resources.

For now, we might lighten the load on that build by removing some optional deps. I'd propose removing matplotlib for a start.

**Comment by TomAugspurger** [comment]
Another OOM crash. I'm removing matplotlib from those two environments.

**Comment by TomAugspurger** [comment]
OK, that passed. Any objections to merging as is, and cleaning up the CI files in a followup PR? I suspect the pytest & pytest-xdist pins are no longer necessary, but it'd be good to get other PRs going again.

**Comment by WillAyd** [comment]
Ok by me

**Comment by jreback** [comment]
hmm I suspect we need to backport this to 0.25.1

**Pull request merged**